doc/classes/TileMapLayer.xml

```diff
@@ -0,0 +1,309 @@
+			</description>
+		</method>
+		<method name="set_cells_terrain_connect">
+			<return type="void" />
+			<param index="0" name="cells" type="Vector2i[]" />
+			<param index="1" name="terrain_set" type="int" />
+			<param index="2" name="terrain" type="int" />
+			<param index="3" name="ignore_empty_terrains" type="bool" default="true" />
+			<description>
+				Update all the cells in the [param cells] coordinates array so that they use the given [param terrain] for the given [param terrain_set]. If an updated cell has the same terrain as one of its neighboring cells, this function tries to join the two. This function might update neighboring tiles if needed to create correct terrain transitions.
+				If [param ignore_empty_terrains] is true, empty terrains will be ignored when trying to find the best fitting tile for the given terrain constraints.
+				[b]Note:[/b] To work correctly, this method requires the TileMapLayer's TileSet to have terrains set up with all required terrain combinations. Otherwise, it may produce unexpected results.
+			</description>
+		</method>
+		<method name="set_cells_terrain_path">
+			<return type="void" />
+			<param index="0" name="path" type="Vector2i[]" />
+			<param index="1" name="terrain_set" type="int" />
+			<param index="2" name="terrain" type="int" />
+			<param index="3" name="ignore_empty_terrains" type="bool" default="true" />
+			<description>
+				Update all the cells in the [param path] coordinates array so that they use the given [param terrain] for the given [param terrain_set]. The function will also connect two successive cell in the path with the same terrain. This function might update neighboring tiles if needed to create correct terrain transitions.
+				If [param ignore_empty_terrains] is true, empty terrains will be ignored when trying to find the best fitting tile for the given terrain constraints.
+				[b]Note:[/b] To work correctly, this method requires the TileMapLayer's TileSet to have terrains set up with all required terrain combinations. Otherwise, it may produce unexpected results.
+			</description>
+		</method>
+		<method name="set_navigation_map">
+			<return type="void" />
+			<param index="0" name="map" type="RID" />
+			<description>
+				Sets a custom [param map] as a [NavigationServer2D] navigation map. If not set, uses the default [World2D] navigation map instead.
+			</description>
+		</method>
+		<method name="set_pattern">
+			<return type="void" />
+			<param index="0" name="position" type="Vector2i" />
+			<param index="1" name="pattern" type="TileMapPattern" />
+			<description>
+				Paste the given [TileMapPattern] at the given [param position] in the tile map.
```

Comment:
I assume the anchor point is the top-left? Unsure how to note it in a few words, but anyway:
```suggestion
				Pastes the [TileMapPattern] at the given [param position] in the tile map. See also [get_pattern].
```